internal/logger/component.go

```diff
@@ -192,10 +204,69 @@ func SerializeConnection(conn Connection, extraKeysAndValues ...interface{}) []i
 		keysAndValues.Add(extraKeysAndValues[i].(string), extraKeysAndValues[i+1])
 	}
 
-	port, err := strconv.ParseInt(conn.ServerPort, 0, 32)
+	port, err := strconv.ParseInt(conn.ServerPort, 10, 32)
 	if err == nil {
 		keysAndValues.Add(KeyServerPort, port)
 	}
 
 	return keysAndValues
 }
+
+// Server contains data that all server messages MAY contain.
+type Server struct {
+	DriverConnectionID uint64             // Driver's ID for the connection
+	TopologyID         primitive.ObjectID // Driver's unique ID for this topology
+	Message            string             // Message associated with the topology
```

Comment:
Including the message here and passing it via the `msg` field in the `LogSink.Info` function seems like it will repeat the message twice in logs. Is that intended?

Reply:
Yes it will. go-logr/logr's LogSink.Info() method requires a message string:

> The log message consists of a constant message attached to the log line. This should generally be a simple description of what's occurring, and should never be a format string.  Variable information can then be attached using named values.

And the DBX logging specifications include the messages in the structured data, e.g. [here](https://github.com/mongodb/specifications/blob/be21d14a95122c35ad45a4e89f07db9c53a00940/source/server-discovery-and-monitoring/server-discovery-and-monitoring-monitoring.rst#server-heartbeat-started-log-message).

The duplication is unfortunate, but I'm not sure there is an obvious way around it. A logrus example:

```
DEBU[0000] Command started                               command="{\"insert\": \"test\",\"ordere..." commandName=insert databaseName=test driverConnectionId=1 message="Command started" operationId=0 requestId=11 serverConnectionId=2014 serverHost=localhost serverPort=27018
```

Reply:
I've added a comment to the "Print" method acknowledging this.

Reply:
I think we have some latitude to interpret the logging specification as requiring passing the message, but not necessarily requiring it to be passed in the key/value pairs. As long as the "default" structured logger output contains the expected structure and a user _can implement_ the expected logger (e.g. by using a structured logger that puts the log message in the "message" field), that should satisfy the specification requirements.